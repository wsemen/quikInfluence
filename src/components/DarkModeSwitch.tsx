import { useColorMode, Switch } from '@chakra-ui/react';

const DarkModeSwitch = () => {
  const { colorMode, toggleColorMode } = useColorMode();
  const isDark = colorMode === 'dark';

  return (
    <Switch
      _focus={{
        borderOutline: 'none',
      }}
      color="green"
      isChecked={isDark}
      onChange={toggleColorMode}
    />
  );
};

export default DarkModeSwitch
