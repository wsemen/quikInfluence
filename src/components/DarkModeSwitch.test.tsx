import { render } from '@testing-library/react'
import DarkModeSwitch from './DarkModeSwitch';

it('renders DarkModeSwitch component', () => {
  const { container } = render(<DarkModeSwitch />)
  expect(container).toMatchSnapshot()
});
