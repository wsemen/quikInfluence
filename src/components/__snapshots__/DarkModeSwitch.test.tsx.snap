// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DarkModeSwitch component 1`] = `
<div>
  <label
    class="chakra-switch css-tm8ajn"
  >
    <input
      class="chakra-switch__input"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-switch__track css-14qxnv8"
    >
      <span
        class="chakra-switch__thumb css-0"
      />
    </span>
  </label>
</div>
`;
